# Template file for 'python3-tornado'
pkgname=python3-tornado
version=6.3.3
revision=1
build_style=python3-pep517
hostmakedepends="python3-setuptools python3-wheel"
makedepends="python3-devel"
depends="python3 ca-certificates"
checkdepends="python3-curl python3-pycares python3-Twisted"
short_desc="Python3 web framework and asynchronous networking library"
maintainer="Orphaned <orphan@voidlinux.org>"
license="Apache-2.0"
homepage="http://www.tornadoweb.org/"
changelog="https://www.tornadoweb.org/en/stable/releases.html"
distfiles="${PYPI_SITE}/t/tornado/tornado-${version}.tar.gz"
checksum=e7d8db41c0181c80d76c982aacc442c0783a2c54d6400fe028954201a2e032fe

do_check() {
	# There is one instance of test_bind_source_ip (the one for
	# curl_httpclient_test.CurlHTTPClientCommonTestCase)
	# which for some reason is taking a little bit over 5s
	# (but only on x86_64-musl and only on CI).
	#
	# The default test timeout is 5s so the test times out and fails.
	# See: https://github.com/void-linux/void-packages/pull/44312#issuecomment-1582361791
	#
	# Setting the timeout to 6 works around it.
	local -x ASYNC_TEST_TIMEOUT=6

	# pytest is not supported
	python3 -m tornado.test
}
